lib/teslamate_web/live/geofence_live/form.html.leex

```diff
@@ -68,7 +68,7 @@
         </p>
         <p class="control is-expanded">
           <%= text_input f, :cost_per_unit, class: "input",
-                             type: :number, inputmode: :decimal, step: 0.0001,
+                             type: :number, inputmode: :decimal, min: 0.0, step: 0.0001,
```

Comment:
Does this remove the ability to set a negative price? Believe it or not, negative pricing is a thing (Octopus Agile in the UK occasionally goes negative). I'm not sure anyone would *want* to set a geofence as negative, but I'm not sure we should explicitly block it?

Reply:
This also potentially explains (some of) the test failures?

Reply:
Good spot on the min.  I will remove to allow users to type in negative numbers.

The test failure is due to the step change.  There is a new test uploaded as part of this change to resolve that.

Reply:
@Ed-M72 Don't the other tests in charge_cost_live_test.exs need updating to reflect the amended precision. I think that's why there are test failures, e.g. https://github.com/adriankumpf/teslamate/pull/1683/checks?check_run_id=3320910767

Reply:
Yes they do.  I have managed to get down to two failed tests, which I can't resolve.  I have mailed Adrian, in the hope he can find some time to pinpoint the reason for the failures as these look like bigger issues I the test script rather than changing a few variables to four decimal places.  Adrian is really busy at the moment so am not optimistic he will find time in the next few months to take a look at it.

Reply:
Fixes are here (no idea how to do a pull request to your pull-request, but hopefully you can see the required changes).

https://github.com/leewillis77/teslamate/commit/a2f56aab94d49f04e6cdbb23d1a08c621e4c67a8

Reply:
The fix is already done to the frontend code.  It is the test scripts which are causing the issue now, and the two failed tests make no sense.  If you get chance would you mind taking look at the failures?

Reply:
The commit I linked to are fixes to the last couple of failing tests?

https://github.com/leewillis77/teslamate/commit/a2f56aab94d49f04e6cdbb23d1a08c621e4c67a8

Reply:
Apologies. I missed the link.  Will take look now.